Assets/SO Architecture/Variables/BaseVariable.cs

```diff
@@ -163,7 +163,7 @@ protected override bool SetValue(T value)
 
             T newValue = _value;
 
-            if (!newValue.Equals(oldValue) && _event != null)
+            if (!EqualityComparer<T>.Default.Equals(_value, newValue))

```

Comment:
It looks like you removed the `_event != null` check, any reason why? I'd add that back in just for safety; I'm not sure what the ramifications for leaving it out would be.

Reply:
Also, shouldn't you be comparing `oldValue` and `newValue`, not `_value` and `newValue`? Looking at the code right above this it looks like `_value` might get changed by `base.SetValue(value)`